### PR TITLE
DOC: Add a link to the migration guide for the deprecation warning of the copy keyword

### PR DESCRIPTION
**Pull request by hmaarrfk**
xref: https://github.com/numpy/numpy/issues/25941#issuecomment-2336522946

### CODE REVIEW

File: numpy/_core/src/multiarray/ctors.c [hunk 1]
```diff
@@ -2476,7 +2476,9 @@ check_or_clear_and_warn_error_if_due_to_copy_kwarg(PyObject *kwnames)
         Py_XDECREF(traceback);
         if (DEPRECATE("__array__ implementation doesn't accept a copy keyword, "
                       "so passing copy=False failed. __array__ must implement "
-                      "'dtype' and 'copy' keyword arguments.") < 0) {
+                      "'dtype' and 'copy' keyword arguments. "
+                      "To learn more, see the migration guide "
+                      "https://numpy.org/devdocs/numpy_2_0_migration_guide.html#adapting-to-changes-in-the-copy-keyword") < 0) {
```
Comment (charris) [inline numpy/_core/src/multiarray/ctors.c:2481]: A 128 character line is hideous :) And there is no guarantee that it will print without breaking.

Comment (hmaarrfk) [reply numpy/_core/src/multiarray/ctors.c:2481]: we can remove the "#" if you want?

I didn't really know there was a migration guide until i asked https://github.com/numpy/numpy/issues/25941#issuecomment-2336522946 and took some of your time...

Comment (charris) [reply numpy/_core/src/multiarray/ctors.c:2481]: Just break the line with quotes. Python combines all quoted bits into one string.

Comment (hmaarrfk) [reply numpy/_core/src/multiarray/ctors.c:2481]: oh you mean hideous in the code, not hideous for the end user.

Comment (hmaarrfk) [reply numpy/_core/src/multiarray/ctors.c]: i made the mod, but i don't think the OSX error is related right?

**Comment by charris** [comment]
Thanks @hmaarrfk .

**Comment by hmaarrfk** [comment]
thank you for the quick review!

**Pull request merged**